vignettes/datatable-programming.Rmd

```diff
@@ -389,6 +389,29 @@ Users will usually want to substitute the function name rather than inserting th
 
 In case of any doubts on the `env` interface functioning, set `verbose = TRUE` to inspect how expressions are resolved internally.
 
+### Use `env` argument from inside another function
+
+It was a design decision that `env` argument will follow _Standard Evaluation_ rules. As a result, it does not require any special handling when being used from inside another function. As well it does not recognize data.table's `.()` alias for a `list()`.
+
+```{r env_se}
+fun = function(x, col.mean) {
+  stopifnot(is.character(col.mean), is.data.table(x))
+  x[, mean(.col), env = list(.col = col.mean)]
+}
+fun(DT, col.mean="Petal.Length")
```

Comment:
```suggestion
fun = function(x, col.mean) {
  stopifnot(is.character(col.mean), is.data.table(x))
  x[, .(col_avg = mean(.col)), env = list(.col = col.mean, . = list)]
}
fun(DT, col.mean="Petal.Length")
```

Reply:
using `. = list` is useless, in i/j/by `.()` will be properly substituted. New commit explains that `env = .(.col="v1")` will not work. `.()` in i/j/by is not affected by `env`

Reply:
Got it, that's good. Thanks for clarifying.